darwin/importer/importer.py

```diff
@@ -267,6 +274,159 @@ def _resolve_annotation_classes(
+            m_prop_type: PropertyType = m_prop.type
+
+            # get metadata property options
+            m_prop_options: List[Dict[str, str]] = m_prop.options or []
+
+            # check if property value is missing for a property that requires a value
+            if m_prop.required and not a_prop.value:
+                _msg = f"Annotation: '{annotation_name}' -> Property '{a_prop.name}' requires a value!"
+                raise ValueError(_msg)
+
+            # check if property and annotation class exists in team
+            if (a_prop.name, annotation_class_id) \
+                not in team_properties_annotation_lookup:
+
+                # if it doesn't exist, create it
+                full_property = FullProperty(
+                    name=a_prop.name,
+                    type=m_prop_type,  # type from .v7/metadata.json
+                    required=m_prop.required,  # required from .v7/metadata.json
+                    description="property-created-during-annotation-import",
+                    slug=client.default_team,
+                    annotation_class_id=int(annotation_class_id),
+                    property_values=[
+                        PropertyValue(
+                            position=m_prop_option.get("position"),  # type: ignore
+                            type=m_prop_option.get("type"),  # type: ignore
+                            value=m_prop_option.get("value"),  # type: ignore
+                            color=m_prop_option.get("color"),  # type: ignore
+                        )
+                        for m_prop_option in m_prop_options  # options from .v7/metadata.json
+                    ],
+                )
+                create_properties.append(full_property)
+                continue
+
+            # check if property value/type is different in m_prop (.v7/metadata.json) options
+            for m_prop_option in m_prop_options:
+                if m_prop_option.get("value") == a_prop.value:
+                    break
+            else:
+                _msg = f"Annotation: '{annotation_name}' -> Property '{a_prop.value}' ({a_prop.type}) not found in .v7/metadata.json, found: {m_prop.options}"
+                raise ValueError(_msg)
+
+            # get team property
+            t_prop: FullProperty = team_properties_annotation_lookup[
+                (a_prop.name, annotation_class_id)
+            ]
+
+            # check if property value is missing for a property that requires a value
+            if t_prop.required and not a_prop.value:
+                _msg = f"Annotation: '{annotation_name}' -> Property '{a_prop.name}' requires a value!"
+                raise ValueError(_msg)
+
+            # check if property value/type is different in t_prop (team) options
+            for t_prop_val in t_prop.property_values or []:
+                if t_prop_val.value == {"value": a_prop.value} and t_prop_val.type == a_prop.type:
+                    break
+            else:
```

Comment:
I actually had to look up the semantics of for-else, I so rarely see them in python.

Reply:
They are quite helpful when we need to know if a for loop had an break statement or not - an alternative to this would be using a extra variable to hold the state :)

> The else clause executes after the loop completes normally. This means that the loop did not encounter a break statement